testing/docker-compose.yml

```diff
@@ -101,6 +101,59 @@ services:
       - "5432:5432"
     volumes:
       - dbdata:/var/lib/postgresql/data
+  ## mock oidc server is configured only for device flow,
+  ## the client_id corresponds to the sda-cli client, not sda-auth
+  oidc:
+    profiles: ["login"]
+    container_name: oidc
+    build:
+      context: ./oidc
+      dockerfile: Dockerfile
+    image: mock-oidc-user-server
+    environment:
+      - PORT=9090
+      - HOST=localhost
+      - CLIENT_ID=sda-cli
+      - CLIENT_REDIRECT_URI=http://localhost:8080/elixir/login
+    ports:
+      - 9090:9090
+    healthcheck:
+      test: ["CMD", "curl", "-f", "http://localhost:9090/.well-known/openid-configuration"]
+      interval: 5s
+      timeout: 10s
+      retries: 4
+  keygen:
+    profiles: ["login"]
+    image: golang:alpine3.16
+    container_name: keygen
+    command:
+      - "/bin/sh"
+      - "-c"
+      - if [ ! -f "/out/c4gh.sec.pem" ]; then wget -qO- "https://github.com/neicnordic/crypt4gh/releases/latest/download/crypt4gh_linux_x86_64.tar.gz" | tar zxf -;
+        ./crypt4gh generate -n c4gh -p privatekeypass && mv *.pem /out/; fi
+    volumes:
+      - /tmp:/out
```

Comment:
Not really a fan of this, yes it is written o `/tmp` but the files are still owned by root since the container doesn't have a `user:` stansa.

Reply:
Not fan of this either. Changed it to a named volume :-)